src/physics/navier_stokes.cpp

```diff
@@ -567,38 +571,72 @@ ::dissipative_flux_templated (
 
 template <int dim, int nstate, typename real>
 void NavierStokes<dim,nstate,real>
-::boundary_face_values (
-   const int boundary_type,
-   const dealii::Point<dim, real> &pos,
+::boundary_wall (
    const dealii::Tensor<1,dim,real> &/*normal_int*/,
-   const std::array<real,nstate> &/*soln_int*/,
-   const std::array<dealii::Tensor<1,dim,real>,nstate> &/*soln_grad_int*/,
+   const std::array<real,nstate> &soln_int,
+   const std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_int,
    std::array<real,nstate> &soln_bc,
    std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_bc) const
 {
-    if (boundary_type == 1000) {
-        // Manufactured solution boundary condition 
-        // -- TO DO: could make this a function later on, similar to how its done in euler
-        // Note: This is consistent with Navah & Nadarajah (2018)
-        std::array<real,nstate> boundary_values;
-        std::array<dealii::Tensor<1,dim,real>,nstate> boundary_gradients;
-        for (int i=0; i<nstate; i++) {
-            boundary_values[i] = this->manufactured_solution_function->value (pos, i);
-            boundary_gradients[i] = this->manufactured_solution_function->gradient (pos, i);
-        }
-        for (int istate=0; istate<nstate; istate++) {
-            soln_bc[istate] = boundary_values[istate];
-            // soln_grad_bc[istate] = soln_grad_int[istate]; // done in convection_diffusion.cpp
-            soln_grad_bc[istate] = boundary_gradients[istate];
-        }
+    using wall_temperature_boundary_condition_enum = Parameters::NavierStokesParam::WallTemperatureBoundaryConditionEnum;
```

Comment:
No need to redefine the "using" if you define it in the header.

Also, I'd recommend renaming mentions of "wall temperature" in the variable/alias names. It might lead to confusions since only isothermal have a prescribed wall temperature.

Reply:
addressed in most recent commit